SatelliteResourceLanguages clarifications for listing multiple languages

I'd like to request two clarifications to the documentation of the SatelliteResourceLanguages property:

1. Change the example to show how to list more than one language, specifically that semicolons must be used as separators.
   <https://github.com/dotnet/docs/blob/ff7e3568e4bfb7283be4bbdb65c2b0024e8169bb/docs/core/project-sdk/msbuild-props.md?plain=1#L386-L390> 
2. Extend the note to explicitly recommend against using the [MSBuild command-line option](https://docs.microsoft.com/visualstudio/msbuild/msbuild-command-line-reference?view=vs-2022) <code>-property:<var>name</var>=<var>value</var></code> to set this property, because it is not possible to list more than one language that way.
   <https://github.com/dotnet/docs/blob/ff7e3568e4bfb7283be4bbdb65c2b0024e8169bb/docs/core/project-sdk/msbuild-props.md?plain=1#L392-L393>

## Context

I tried to set SatelliteResourceLanguages on the `dotnet publish` command line, using .NET SDK 6.0.302 on Windows:

```
PS> dotnet publish -o "$(Get-Location)/bin/net6.0" --no-self-contained --configuration Release --framework net6.0 -p:SatelliteResourceLanguages="en;de"
Microsoft (R) Build Engine version 17.2.0+41abc5629 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1006: Property is not valid.
Switch: de

For switch syntax, type "MSBuild -help"
```

It didn't work, because the `-p` option parsed the semicolon as terminating the property value and starting another property name, regardless of the quotation marks.
Next, I tried encoding the semicolon as `%3B`, as listed in [MSBuild special characters](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-special-characters?view=vs-2019):

```
PS> dotnet publish -o "$(Get-Location)/bin/net6.0" --no-self-contained --configuration Release --framework net6.0 -p:SatelliteResourceLanguages="en%3Ade"
Microsoft (R) Build Engine version 17.2.0+41abc5629 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored C:\MyApp\src\MyApp\MyApp.csproj (in 542 ms).
  MyApp -> C:\MyApp\src\MyApp\bin\Release\net6.0\MyApp.dll
  MyApp -> C:\MyApp\bin\net6.0\
```

That appeared to succeed, but .NET SDK did not copy the "de" satellite assembly to C:\MyApp\bin\net6.0\de\System.CommandLine.resources.dll. I think the reason is that [the SatelliteResourceLanguages parameter of the ResolvePackageAssets task](https://github.com/dotnet/sdk/blob/dc262f0c444d57181e580d4d5b9a517ea558c64a/src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs#L77) is defined as `ITaskItem[]` rather than `string`, so when [the ResolvePackageAssets target passes $(SatelliteResourceLanguages) to that parameter](https://github.com/dotnet/sdk/blob/dc262f0c444d57181e580d4d5b9a517ea558c64a/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.PackageDependencyResolution.targets#L290), MSBuild has to split the string value to items, but if the semicolon was encoded, then MSBuild doesn't treat it as a list separator.

Finally, I set the property in Directory.Build.props, to make it apply to all projects in the solution

```XML
<Project>
  <PropertyGroup>
    <SatelliteResourceLanguages>en;de</SatelliteResourceLanguages>
  </PropertyGroup>
</Project>
```

and ran `dotnet publish` without the <code>-property:SatelliteResourceLanguages=<var>value</var></code> option, and that worked fine.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: cb4428fc-215c-78e1-5ff0-fe73f7a08260
* Version Independent ID: 6741b1de-38a4-fbb1-d79d-acc47d549a2d
* Content: [MSBuild properties for Microsoft.NET.Sdk - .NET](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#satelliteresourcelanguages)
* Content Source: [docs/core/project-sdk/msbuild-props.md](https://github.com/dotnet/docs/blob/main/docs/core/project-sdk/msbuild-props.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

--- a/docs/core/project-sdk/msbuild-props.md
+++ b/docs/core/project-sdk/msbuild-props.md
@@ -381,11 +381,11 @@ The `RuntimeIdentifiers` property lets you specify a semicolon-delimited list of
 
 ### SatelliteResourceLanguages
 
-The `SatelliteResourceLanguages` property lets you specify which languages you want to preserve satellite resource assemblies for during build and publish. Many NuGet packages include localized resource satellite assemblies in the main package. For projects that reference these NuGet packages that don't require localized resources, the localized assemblies can unnecessarily inflate the build and publish output size. By adding the `SatelliteResourceLanguages` property to your project file, only localized assemblies for the languages you specify will be included in the build and publish output. For example, in the following project file, only English (US) resource satellite assemblies will be retained.
+The `SatelliteResourceLanguages` property lets you specify which languages you want to preserve satellite resource assemblies for during build and publish. Many NuGet packages include localized resource satellite assemblies in the main package. For projects that reference these NuGet packages that don't require localized resources, the localized assemblies can unnecessarily inflate the build and publish output size. By adding the `SatelliteResourceLanguages` property to your project file, only localized assemblies for the languages you specify will be included in the build and publish output. For example, in the following project file, only English (US) and German (Germany) resource satellite assemblies will be retained.
 
 ```xml
 <PropertyGroup>
-  <SatelliteResourceLanguages>en-US</SatelliteResourceLanguages>
+  <SatelliteResourceLanguages>en-US;de-DE</SatelliteResourceLanguages>
 </PropertyGroup>
 ```
 

